osu.Framework.Tests/Visual/Sprites/TestSceneAnimation.cs

```diff
@@ -249,13 +250,22 @@ public void TestGotoFrameBeforeLoaded(int frame)
         [Test]
         public void TestClearFrames()
         {
+            Texture lastFrame = null;
+
             loadNewAnimation();
 
             AddUntilStep("animation is playing", () => animation.CurrentFrameIndex > 0);
 
+            AddStep("store current frame", () => lastFrame = animation.CurrentFrame);
+
             AddStep("clear frames", () => animation.ClearFrames());
+
             AddAssert("animation duration is 0", () => animation.Duration == 0);
             AddAssert("animation is at start", () => animation.CurrentFrameIndex == 0);
+            AddAssert("animation is not showing frame", () => animation.ChildrenOfType<Sprite>().First().Texture == null);
+
+            AddStep("add one frame back", () => animation.AddFrame(lastFrame));
+            AddAssert("animation is showing frame", () => animation.ChildrenOfType<Sprite>().First().Texture != null);
```

Comment:
A bit of a weak assertion this, to the point where it doesn't actually fail for me with the fixes reverted. Any reason not to do the following?

```suggestion
            AddAssert("animation is showing frame", () => animation.ChildrenOfType<Sprite>().First().Texture == lastFrame);
```